src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -99,7 +108,11 @@ public IPipeWriter CreateWriter(IPipeWriter writer, Func<IPipeReader, IPipeWrite
         public IPipeReader CreateReader(IPipeReader reader, Func<IPipeReader, IPipeWriter, Task> produce)
```

Comment:
Yes. The idea is to not expose the pipe but now that we have `IPipe` it's conceivable that most of these methods just go away.